src/ToDo/Data/Mock/MockTaskListEndpoint.cs

```diff
@@ -142,5 +144,54 @@ public async Task<TaskReponseData<TaskData>> GetAllTasksAsync(string displayName
 		return new TaskReponseData<TaskData> { Value = tasks.Where(x => x.Title != null && x.Title.Contains(displayName)).ToList() };
 	}
 
+	public async Task<CheckListItemData> AddStepAsync(string baseTaskListId, string baseTaskId,
+		CheckListItemData checkListItem, CancellationToken ct)
+	{
+		var list = await LoadListTasks(baseTaskListId);
+		var task = list.FirstOrDefault(t => t.Id == baseTaskId);
+		if (task is not null)
+		{
+			if(task.ChecklistItems is null)
+			{
+				task.ChecklistItems = new List<CheckListItemData>();
+			}
 
+			task.ChecklistItems?.Add(checkListItem);
+			return checkListItem;
+		}
+		return new CheckListItemData();
+	}
+
+	public async Task<TaskReponseData<CheckListItemData>> GetStepsAsync(string baseTaskListId, string baseTaskId, CancellationToken ct)
+	{
+		var list = await LoadListTasks(baseTaskListId);
+		var task = list.FirstOrDefault(t => t.Id == baseTaskId);
+		if (task is not null)
```

Comment:
combine with previous line eg
if (list.FirstOrDefault(t => t.Id == baseTaskId) is { } task)